dask/dataframe/core.py

```diff
@@ -1090,6 +1101,55 @@ def _aca_agg(self, token, func, aggfunc=None, **kwargs):
                    columns=return_scalar, token=self._token_prefix + token,
                    **kwargs)
 
+    def apply_concat_apply(self, func, aggfunc=None, func_kwargs=None,
+                           aggfunc_kwargs=None):
+        """ Apply a function to each partition, the concat, then apply again
+
+        Useful to perform easily parallelizable aggregations based on data from
+        the whole series, like minimum, maximum, mean, etc.
+
+        Parameters
+        ----------
+
+        func: function [partition-per-arg] -> partition
+            Function to operate on each partition
+        aggfunc: function concatenated-partition -> partition
+            Function to operate on the concatenated result of func. Defaults to
```

Comment:
Hmmm I believe in most cases the result of applying an aggregation function to a DataFrame or Series will be another DataFrame or Series, therefore `_concat` should work as `pd.concat`. Which cases can you think of that wouldn't behave like this?

Perhaps because of this we should think of renaming the function to `aggregate` or `aggregate_partitioned` to emphasize `func` and `aggfunc` are both expected to be aggregation functions.